core/src/main/java/com/github/philippheuer/events4j/core/EventManager.java

```diff
@@ -60,9 +57,14 @@
     /**
      * Default EventHandler
      */
-    @Setter
     private String defaultEventHandler;
 
+    /**
+     * Holds all active subscriptions
+     */
+    @Getter
+    private final List<IEventSubscription> activeSubscriptions = Collections.synchronizedList(new ArrayList<>());
```

Comment:
Is it worth changing this to `Map<String, List<IEventSubscription>>` (map from consumer id to consumers with that id) so [line 231](https://github.com/PhilippHeuer/events4j/pull/8/files#diff-eb91e0437b564d068d5894cc4cdad6bd26f7aaab597efbda04af0cb6846ca39cR231) can be `O(1)` instead of `O(n)`?

Would increase code complexity and have memory overhead because there could be a lot of one-element lists. At the same time, would avoid locking when there are concurrent mutations on different consumer id's whereas the current synchronizedList would always block

Reply:
How about a approach thats kind of in the middle, ie. we use a `Map<String, IEventSubscription>` so we have `O(1)`.

Then id's have to be unqiue, which means we have to add a incrementing seqeuence to the end of the id if it's already present unless `onEventIfIdUnique` is used.

Reply:
I do like the simplification - only this is we might need to keep a `Map<String, Integer>` to solve the following situation:

1. onEvent is called with id: `e`
2. onEvent is called with the same id, `e`. This one is stored in the map as `e-1`.
3. The first onEvent is disposed so there is no longer a mapping for `e`.
4. onEventIfUnique is called with id: `e`. It needs to recognize that `e-1` is registered so uniqueness is false

So, we may have to keep a running counter for the original id & the number of subscriptions to properly determine uniqueness

Reply:
Thats a good point. I don't think that its worth it to build all that for our use-case which is quite simple.

I changed the onEvent method to return `IEventSubscription` which is the `IDisposable` along with the event type + actual id.

How about this:
- user provided name/id's have to be unique
- if no name is set, the library generates a random unique name that will be used internally (accessible via `IEventSubscription.getId()` when registering a consumer)

That way we could avoid all the issues above.

Reply:
I like that approach best - avoids needless complexity and is lock-free (if we were doing the two map approach, we may have needed locking on writes to keep them in sync)